src/Microsoft.Data.Sqlite/Interop/MarshalEx.cs

```diff
@@ -60,11 +61,27 @@ public static void ThrowExceptionForRC(int rc, Sqlite3Handle db)
                 return;
             }
 
-            var message = db == null || db.IsInvalid
-                ? NativeMethods.sqlite3_errstr(rc)
-                : NativeMethods.sqlite3_errmsg(db);
 
-            throw new SqliteException(Strings.FormatSqliteNativeError(rc, message), rc);
+            string message;
+
+            if (db != null && !db.IsInvalid)
+            {
+                message = Strings.FormatSqliteNativeError(
+                            rc,
+                            NativeMethods.sqlite3_errmsg(db));
+            }
+            else if (SqliteVersion.Current >= "3.7.15")
```

Comment:
Careful, the string "3.10" is less than "3.7". May want to use the `Version` class.